lamellar/lamellar_lsbsort/src/am_safe.rs

```diff
@@ -0,0 +1,363 @@
+#![allow(non_snake_case)]
+use rand::distr::StandardUniform;
+use rand::prelude::*;
+use rand_pcg::Pcg64;
+use std::sync::Mutex;
+use std::time::Instant;
+
+use lamellar::active_messaging::prelude::*;
+use lamellar::array::prelude::*;
+use lamellar::array::Distribution;
+use lamellar::darc::prelude::*;
+
+const RADIX: u64 = 16; // TODO set to 16
+const N_DIGITS: u64 = 64 / RADIX;
+const N_BUCKETS: u64 = 1 << RADIX;
+const MASK: u64 = N_BUCKETS - 1;
+
+// An Active message based approach to LSB Radix Sort
+// Each PE is responsible for N_BUCKETS/num_pes buckets
+// data is sorted into buckets based on the current digit.
+// At the end of the sort, data is evenly re-distributed across the pes.
+// During the sort it is possible to become unbalanced
+// based on the distribution of the keys. For uniform data
+// this shouldnt be too bad.
```

Comment:
I am happy to include this implementation in this repository, but I don't consider it the same algorithm as the others, because it doesn't do a global shuffle for each digit, instead there is just one global shuffle. So I wouldn't compare it directly, but rather have it as an alternative implementation for context.